bdk_chain/src/keychain.rs

```diff
@@ -20,13 +20,79 @@ mod keychain_txout_index;
 #[cfg(feature = "miniscript")]
 pub use keychain_txout_index::*;
 
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Deserialize, serde::Serialize),
+    serde(
+        crate = "serde_crate",
+        bound(
+            deserialize = "K: Ord + serde::Deserialize<'de>",
+            serialize = "K: Ord + serde::Serialize"
+        )
+    )
+)]
+#[must_use]
```

Comment:
Does this work when the thing is inside a tuple?

Reply:
The `#[must_use]`? Or the derives?

Reply:
I assume so...